siliconcompiler/client.py

```diff
@@ -309,17 +309,25 @@ def fetch_results_request(chip):
             elif resp.status_code == 303:
                 redirect_url = resp.headers['Location']
                 can_redirect = True
-            else:
+            elif resp.status_code == 200:
                 shutil.copyfileobj(resp.raw, zipf)
-                return
+                return 0
+            elif int(resp.status_code / 100) in [4, 5]:
+                msg = '.'
+                try: # (An unexpected server error may not return JSON with a message)
+                    msg = f': {resp.json()["message"]}'
+                except:
```

Comment:
except what? enumerate and generate appropriate messages

Reply:
The comment explains: the exception may occur if the response is not in a JSON format. It is expected to be, but the server could also encounter an unexpected error and return a 500 response without content.

Reply:
Replaced the raw `except:` with a specific exception.